pallav/fashionTech/pallavBanka/parkingAssignment/Parking.java

```diff
@@ -0,0 +1,122 @@
+package fashionTech.pallavBanka.parkingAssignment;
+
+import java.util.*;
+
+class ParkingCharge {
+    Date inTime;
+    Date outTime;
+    private Long ParkingTime(Date inTime, Date outTime) {
+        this.inTime = inTime;
+        this.outTime = outTime;
+        if (this.outTime != null) {
+            //difference in time in milliseconds
+            Long diff = outTime.getTime() - inTime.getTime();
+            Long diffHours = diff / (60 * 60 * 1000) % 24;
+            return diffHours;
+        } else{
+            System.out.println("The vehicle hasn't left the premises.");
+            return null;
+        }
+    }
+    private Long charges(String s){
+        Hashtable<String, Integer> charges=new Hashtable<String, Integer>();
+        charges.put("TwoWheeler", 20);
+        charges.put("FourWheeler", 40);
+        charges.put("SixWheeler", 80);
+        return charges.get(s)*ParkingTime(inTime, outTime);
+    }
+}
+
+class Floor {
+    private void floor() {
+        Hashtable<Integer, String> vehiclewisefloor = new Hashtable<Integer, String>();
+        vehiclewisefloor.put(0,"SixWheeler");
+        vehiclewisefloor.put(1, "TwoWheelers");
+        vehiclewisefloor.put(2, "TwoWheelers");
+        vehiclewisefloor.put(3,"FourWheeler");
+        vehiclewisefloor.put(4,"FourWheeler");
+        vehiclewisefloor.put(5, "TwoWheelers");
+    }
+}
+
+class Slot{
+    ArrayList<Integer> unoccupied=new ArrayList<Integer>();
```

Comment:
Use LinkedList, as you will be constantly inserting and updating this list.
List<Integer> unoccupiedSlots = new LinkedList();
Also, I guess for occupied scenario, you can maintain a Map with regNo as key and slotId as value, so that when you get a vehicle from parking, you can put that slotId back in unoccupied List.